### PR TITLE
Fix no json parsing check

### PR DESCRIPTION
**Pull request by charlesfleche**
The ObjectLoader didn't check for JSON parsing errors : if such an error happened, the onError callback wouldn't be called. This patch encloses the JSON parsing in try / catch block that calls onError in case of error.

### CODE REVIEW

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -53,7 +53,24 @@ Object.assign( ObjectLoader.prototype, {
...
+			var json = null;
+			
+			try {
+
+				json = JSON.parse( text );
+
+                        } catch( error ) {
+
+				console.log(err.message);
+				onError(loader);
```
Comment (charlesfleche) [inline src/loaders/ObjectLoader.js]: My bad, I mistook XHRLoader for XMLHttpRequest. At that point the XMLHttpRequest succeeded and is not accessible anymore. Should we call onError without parameters ?

**Comment by mrdoob** [comment]
Implemented in the `dev` branch. Thanks!

**Comment by charlesfleche** [comment]
Thanks ! However, it seems that the onError callback is not called in case of JSON parsing error. Is it really the desired behavior ?

**Comment by sandromartis** [comment]
I think I agree with @charlesfleche here. In my opinion the `onError` callback should be called when there is a parsing error, or some other error is thrown.

I am using the `ObjectLoader` in a worker queue and when the workers don't get a response from the `ObjectLoader` through the `onLoad` or `onError` callbacks they get stuck and cannot continue loading additional files.

**Comment by mrdoob** [comment]
Sounds good!

**Comment by mrdoob** [comment]
Done!

**Pull request closed without merging**